### ISSUE TITLE
Stop shipping unit tests with the extension

### ISSUE THREAD

**Issue by ghostwords** [EFForg/privacybadger#1647]
To have the published extension go from 1.5M to 855K.

Requires updating release scripts (probably) and documentation (possibly, to make sure it's clear you need Privacy Badger from source code in order to run unit tests).

**Comment by ghostwords** [comment]
Follows up on #1646.

**Comment by ghostwords** [comment]
Previously: https://github.com/EFForg/privacybadger/pull/1406#issuecomment-305514523.

**Comment by cowlicks** [comment]
I think shipping the tests are ok, they are small. Sinon, and Qunit are less small. I think this is a good case for having them installed via our package.json, and out of git. It should make building easier since we won't have to manually remove them from the repo.

Re: https://github.com/EFForg/privacybadger/pull/1250#issuecomment-286632862

Also, more discussion is here https://github.com/EFForg/privacybadger/pull/1278#issuecomment-289970073

**Comment by ghostwords** [comment]
What's the point of shipping tests when they won't work unless you start running `npm` commands inside the browser's extensions folder (not recommended: folder hard to find, contents could get reset at any point by the browser, ...)?

**Comment by cowlicks** [comment]
The point of shipping the testing files, and not the dependencies, would be to make the build process simpler, and therefore less error prone, and easier to maintain.

If someone wants to run the unittests,  I would suggest they follow the developer instructions for installing privacy badger. I would not suggest that anyone run `npm` inside the browser folder, that is indeed a bad idea.

If we used the unittests to debug normal (from app store) installations of privacy badger, then having the tests be immediately runnable without npm would be useful. But, to my knowledge, we've never used unittests to debug normal PB installations. Have we? Was it useful?

**Comment by ghostwords** [comment]
Right, we haven't, which is why they should be removed from the shipped extension package.

**Comment by ghostwords** [comment]
I don't think removing tests from the repo in release scripts (probably) is something we need to worry about avoiding. The benefit (almost halving the package size) seems to easily outweigh the cost (the one extra line to exclude a subdirectory).

**Comment by cowlicks** [comment]
Having the test dependencies managed by `npm` would also let us upgrade them more easily. And therefore be able to receive upstream bug fixes more easily. So if Sinon fixed the bugs we saw in #1618 we be able to use the fixes.

**Comment by ghostwords** [comment]
Wait a second, we already remove unit tests from releases: https://github.com/EFForg/privacybadger/blob/19d6263e16ceeabe461dc3b546f6e23fe792511d/release-utils/make-eff-zip.sh#L17.

```bash
$ wget https://www.eff.org/files/privacy-badger-latest.xpi
$ du -sh privacy-badger-latest.xpi
836K    privacy-badger-latest.xpi
```
